compiler/src/main/scala/dx/compiler/WorkflowCompiler.scala

```diff
@@ -569,7 +569,7 @@ case class WorkflowCompiler(separateOutputs: Boolean,
                   if (app.kind.getClass == classOf[ExecutableKindWfFragment]) =>
                 "do not override"
               case static: StaticInstanceType => instanceTypeDb.apply(static.req).name
-              case DefaultInstanceType if (app.kind.getClass == classOf[ExecutableKindNative]) =>
+              case DefaultInstanceType =>
```

Comment:
https://github.com/dnanexus/dxCompiler/blob/c7b3061e49f5e7f413e0481fd880a8393655a256/compiler/src/main/scala/dx/translator/wdl/CallableTranslator.scala#L684-L701

So here the frag applet should always have the same instance type as its callee, and if the frag is wrapping a native applet with `DefaultInstanceType`, its own instance type should also be preserved, right? @Gvaihir

Reply:
The frag applet should always run on a small default instance. What happens here is:
1. Frag is translated with the same instance type as callee.
2. In [`ApplicationCompiler`](https://github.com/dnanexus/dxCompiler/blob/7592ed8d566a31b9438032596f737af69e4788f2/compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala#L171) it gets overridden by specifying a default small instance from `InstanceTypeDb` for current project it's `mem1_ssd1_v2_x2`.
3. In the [`WorkflowCompiler`](https://github.com/dnanexus/dxCompiler/blob/9297114c9d9811611dc0523522e7377e0ca26a9e/compiler/src/main/scala/dx/compiler/WorkflowCompiler.scala#L564) the original instance type of a frag is passed to the `systemRequirements` of the native callee to override the default instance. So the original instance type of the frag is just preserved for its child process. 

It's like a trust fund - you put money there but only your (grand)children can withdraw, on a very high level lol :)

For the compiled execs it make no sense because you don't override instances - you compile with whatever you specify (or don't specify).

Reply:
Thanks for the explanation! The metaphor is awesome :D